### PR TITLE
[OptApp] NAG method

### PR DESCRIPTION
**Pull request by Igarizza**
**📝 Description**
Adding new optimization method for unconstrained problems.

### CODE REVIEW

File: applications/OptimizationApplication/python_scripts/algorithms/algorithm_nesterov_accelarated_gradient.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+from KratosMultiphysics.OptimizationApplication.algorithms.algorithm_steepest_descent import AlgorithmSteepestDescent
+
+
+def Factory(model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+    return AlgorithmNesterovAcceleratedGradient(model, parameters, optimization_problem)
+
+class AlgorithmNesterovAcceleratedGradient(AlgorithmSteepestDescent):
+    """
+        A classical steepest descent algorithm to solve unconstrainted optimization problems.
+    """
```
Comment (sunethwarna) [inline applications/OptimizationApplication/python_scripts/algorithms/algorithm_nesterov_accelarated_gradient.py]: Documentation. I would add publication references as well.

File: applications/OptimizationApplication/python_scripts/algorithms/algorithm_nesterov_accelarated_gradient.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        self.prev_update = None
+        self.eta = self.parameters["settings"]["eta"].GetDouble()
+
+    @time_decorator()
+    def ComputeControlUpdate(self, alpha):
+        search_direction = self.algorithm_data.GetBufferedData()["search_direction"]
+        if isinstance(alpha, float):
+            update = search_direction * alpha
+        elif isinstance(alpha, KratosOA.CollectiveExpression):
+            update = search_direction.Scale(alpha)
```
Comment (sunethwarna) [inline applications/OptimizationApplication/python_scripts/algorithms/algorithm_nesterov_accelarated_gradient.py]:
```suggestion
        update = Kratos.Expression.Utils.Scale(search_direction, alpha)
```

**Review by sunethwarna — COMMENTED** [review]
I only have minor comments. Otherwise, this is cool :)

File: applications/OptimizationApplication/python_scripts/algorithms/algorithm_nesterov_accelarated_gradient.py (hunk 1 above)
Comment (Igarizza) [reply applications/OptimizationApplication/python_scripts/algorithms/algorithm_nesterov_accelarated_gradient.py]: You mean class description or I should add it to oyur documentation?

Comment (Igarizza) [reply applications/OptimizationApplication/python_scripts/algorithms/algorithm_nesterov_accelarated_gradient.py]: I have added class description with paper link

File: applications/OptimizationApplication/python_scripts/algorithms/algorithm_nesterov_accelarated_gradient.py (hunk 2 above)
Comment (Igarizza) [reply applications/OptimizationApplication/python_scripts/algorithms/algorithm_nesterov_accelarated_gradient.py]: done

**Review by sunethwarna — APPROVED** [review]
Thanks @Igarizza.

**Comment by sunethwarna** [comment]
No idea... But we are not touching it, so it should be safe.

**Pull request merged**